src/psyclone/tests/psyir/nodes/call_test.py

```diff
@@ -418,6 +421,31 @@ def test_call_reference_accesses():
     assert not var_info.is_written(Signature("beta"))
 
 
+def test_type_bound_call_reference_accesses(fortran_reader):
+    '''Test the reference_accesses() method for a call to a type-bound
+    procedure.'''
+    code = '''\
+    subroutine my_sub
+      use some_mod, only: my_grid
+      integer :: i, j, k
+      call my_grid(k)%update(i,j)
```

Comment:
Maybe one of the indices should be another type-bound call with more indices, to check that it compounds properly.